build/page-description.ts

```diff
@@ -0,0 +1,83 @@
+import { DEFAULT_LOCALE } from "../libs/constants/index.js";
+import { Doc } from "../libs/types/document.js";
+
+export function getPageDescription(
+  doc: Partial<Doc>,
+  pageType: string
+): string | undefined {
+  if (doc.locale === DEFAULT_LOCALE) {
+    const sections = doc.toc?.map(({ text }) => text) ?? [];
+
+    const syntaxItems = Object.entries({
+      Value: "type",
+      "Event type": "type",
+      Syntax: "syntax",
+      Parameters: "parameters",
+      Constructor: "constructor",
+      "Instance properties": "properties",
+      "Event properties": "properties",
+      "Instance methods": "methods",
+      "Return value": "return value",
+    })
+      .filter(([section]) => sections.includes(section))
+      .map(([, text]) => text);
+
+    const otherItems = Object.entries({
+      Exceptions: "exceptions",
+      Examples: "code examples",
+      Specifications: "specifications",
+      "Browser compatibility": "browser compatibility",
+    })
+      .filter(([section]) => sections.includes(section))
+      .map(([, text]) => text);
+
+    const syntaxContents = enumerate(...syntaxItems);
+    const otherContents = enumerate(...otherItems);
+    const contents = [
+      syntaxContents ? `its ${syntaxContents}` : "",
+      otherContents,
+    ]
+      .filter(Boolean)
+      .join(", ");
```

Comment:
e.g. "its properties, and methods, code examples, specifications, and browser compatibility"